### PR TITLE
[Salesforce] Advanced Sync Rules documentation

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Relates to https://github.com/elastic/connectors/issues/1859

Documentation for Salesforce Advanced Sync Rules.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

**Comment by leemthompo** [comment]
@moxarth-elastic you can use the comments on https://github.com/elastic/connectors/pull/1936 as a guide to make the headings, formatting, etc. clearer

File: docs/sync-rules/SALESFORCE.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+### Setting up the Dropbox connector
```
Comment (seanstory) [inline docs/sync-rules/SALESFORCE.md:1]:
```suggestion
### Setting up the Salesforce connector
```

**Review by wangch079 — COMMENTED** [review]
I see `Indexing` and `fetch` are used interchangeably. The rules are actually applied when fetching data. Can we make it consistent?

**Comment by moxarth-elastic** [comment]
> I see `Indexing` and `fetch` are used interchangeably. The rules are actually applied when fetching data. Can we make it consistent?

Yes, i've made it consistent.

**Review by navarone-feekery — APPROVED** [review]
LGTM!

**Comment by moxarth-elastic** [comment]
@seanstory @wangch079 your comments are addressed, can you guys please take a look? We need one more approval to merge this PR.

**Pull request merged**